hilti/toolchain/src/ast/operators/bytes.cc

```diff
@@ -876,6 +876,29 @@ byte order, and converts it into an unsigned integer.
 };
 HILTI_OPERATOR_IMPLEMENTATION(ToUIntBinary);
 
+class ToRealAscii : public BuiltInMemberCall {
+public:
+    Signature signature(Builder* builder) const final {
+        return Signature{
+            .kind = Kind::MemberCall,
+            .self = {parameter::Kind::In, builder->typeBytes()},
+            .member = "to_real",
+            .result = {Constness::Const, builder->typeReal()},
+            .ns = "bytes",
+            .doc =
+                R"(
+Interprets the ``bytes`` as representing an ASCII-encoded floating point number
+and converts that into a ``real``. The data can be in either decimal or
+hexadecimal format. If it cannot be parsed as either, throws an `InvalidValue`
+exception.
```

Comment:
All of this depends on the locale (decimal separator), could you add something calling that out?

Reply:
This is a good one. I don't think we want it to depend on locale, that could end up being very surprising. From a look it seems hard to come up with version that doesn't unfortunately, but I'll look around a bit more.